opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanBuilderShim.java

```diff
@@ -232,4 +254,69 @@ public Span start() {
 
     return new SpanShim(telemetryInfo(), span, baggage);
   }
+
+  // The first SpanContext with Child Of type in the entire list is used as parent,
+  // else the the first SpanContext is used as parent.
+  @Nullable
+  static io.opentelemetry.api.trace.SpanContext getMainParent(List<SpanParentInfo> parents) {
+    if (parents.size() == 0) {
+      return null;
+    }
+
+    SpanParentInfo mainParent = null;
+
+    for (SpanParentInfo parentInfo : parents) {
+      if (parentInfo.getRefType() == ReferenceType.CHILD_OF) {
+        mainParent = parentInfo;
+        break;
+      }
+    }
+
+    mainParent = mainParent == null ? parents.get(0) : mainParent;
+    return mainParent.getSpanContext();
+  }
+
+  static Baggage getAllBaggage(List<SpanParentInfo> parents) {
+    if (parents.size() == 0) {
+      return Baggage.empty();
+    }
+
+    if (parents.size() == 1) {
+      return parents.get(0).getBaggage();
+    }
+
+    BaggageBuilder builder = Baggage.builder();
+    for (SpanParentInfo parent : parents) {
+      parent
+          .getBaggage()
+          .forEach(
+              (key, entry) -> {
+                builder.put(key, entry.getValue());
+              });
```

Comment:
#nit: No need for braces since it's a single statement.

Reply:
Updated.